### PR TITLE
Issue 6853: Expose location details of a segment.

### PR DESCRIPTION
**Pull request by anju-c-das**
Signed-off-by: anju_das <anju.das@dell.com>

**Change log description**  
Added a new method in ClientConnection.java to expose the location/endpoint details of a segment. 

**Purpose of the change**  
Fixed https://github.com/pravega/pravega/issues/6853

**How to verify it**  
All test cases should pass

### CODE REVIEW

File: client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java [hunk 1]
```diff
@@ -69,4 +70,9 @@ interface CompletedCallback {
         void complete(ConnectionFailedException e);
     }
 
+    /**
+     * Get the endpoint details of a segment.
```
Comment (ShwethaSNayak) [inline client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java:74]: description can be changed to "Get the location details of a segment" to be in consistent with other terms in the file.

**Comment by ShwethaSNayak** [comment]
Did you try with having default method to get the location, instead of implementing them in all impl classes?

File: client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java (hunk 1 above)
Comment (anju-c-das) [reply client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java]: Incorporated the review comment.

**Comment by anju-c-das** [comment]
> Did you try with having default method to get the location, instead of implementing them in all impl classes?

Incorporated changes

File: client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java [hunk 2]
```diff
@@ -26,6 +27,9 @@
  */
 public interface ClientConnection extends AutoCloseable {
 
+
+    PravegaNodeUri NODE_LOCATION = null;
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java:31]: I don't think this field is needed. It's fine for the interface to simply provide the method signature and leave it up to the implementations.

File: client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java [hunk 3]
```diff
@@ -70,4 +71,10 @@ public void close() {
         }
     }
 
+    @Override
+    public PravegaNodeUri getLocation() { 
+        String[] nodeLocation = connectionName.split(":");
+        return new PravegaNodeUri(nodeLocation[0].trim(), Integer.parseInt(nodeLocation[1].trim()));
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java:77]: Don't add parsing logic here. We already have utility functions for this in `shared`. If you can't find what you want in `NameUtils` maybe add it there.

**Review by tkaitchuck — CHANGES_REQUESTED** [review]

File: client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java (hunk 3 above)
Comment (anju-c-das) [reply client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java]: Incorporated changes

File: client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java (hunk 2 above)
Comment (anju-c-das) [reply client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java]: Incorporated changes

File: client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java [hunk 4]
```diff
@@ -69,4 +70,11 @@ interface CompletedCallback {
         void complete(ConnectionFailedException e);
     }
 
+    /**
+     * Get the location details of a segment.
+     */
+    default PravegaNodeUri getLocation() {
+        return null;
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/connection/impl/ClientConnection.java:77]: We don't need a default impl here.

**Review by tkaitchuck — APPROVED** [review]

File: client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java [hunk 5]
```diff
@@ -70,4 +73,10 @@ public void close() {
         }
     }
 
+    public PravegaNodeUri getLocation() {
```
Comment (RaulGracia) [inline client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java:76]: Do we need javadoc here?

**Review by RaulGracia — COMMENTED** [review]
Looks good. Please, change the PR description to describe how this PR has been tested (currently it says N/A, which is not true).

File: client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java (hunk 5 above)
Comment (anju-c-das) [reply client/src/main/java/io/pravega/client/connection/impl/FlowClientConnection.java:76]: Added Java doc for same.

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**